### ISSUE TITLE
Create Cross-project Logging Sink throws errors

### ISSUE THREAD

**Issue by pdanysz** [googleapis/nodejs-logging#684]
#### Problem
I tried today to automate some stuff in cloud with NodeJS and Logging SINK create feature across many project I have (let say I want to centralize some events from logs in one project).

I fallowed the docs and create something like that:
```
async function createLogsExportToPubSub( projectId ) {

    const logging = new Logging({
        projectId: projectId
    });

    const config = {
        uniqueWriterIdentity: true,
        destination: `pubsub.googleapis.com/projects/${PROJECT_ID}/topics/${SINK_TOPIC}`,
        filter: fs.readFileSync('only-errors.filter').toString(),
    }

    const sink = logging.sink( SINK_NAME );

    sink.create( config ).then( status => {
        console.log('Sink created', sink);
    }, err => {
        console.error('Error while creating a sink', err, err.message);
    });
}
```
I configure a new Service Account with proper permissions (Logs Configuration Writer) and setup proper settings in "config" for sink creation `` uniqueWriterIdentity: true``

Unfortunately this code won't work due to the bug? I suppose in library code.
When you run this code you will get error about "uniqueWriterIdentity" was not set:
```
Error: 3 INVALID_ARGUMENT: The destination project must be the same as the sink's project. Use uniqueWriterIdentity=true to allow cross project exports.
```

So, I started to dig a litle bit in this library and I found some in this file showed below.
It looks like that property ``uniqueWriterIdentity`` is never been used by constructors of requests and that probably cause the problem.

As WA I added to the ``reqOpts`` object a property ``uniqueWriterIdentity: true`` and then re-run my code :), that helps, sink was created in other project as I wanted.

#### Library
node_modules/@google-cloud/logging/build/src/index.js 
start around line nubmer 145
```
async createSink(name, config) {
        if (typeof name !== 'string') {
            throw new Error('A sink name must be provided.');
        }
        if (typeof config !== 'object') {
            throw new Error('A sink configuration object must be provided.');
        }
        if (common_1.util.isCustomType(config.destination, 'bigquery/dataset')) {
            await this.setAclForDataset_(config);
        }
        if (common_1.util.isCustomType(config.destination, 'pubsub/topic')) {
            await this.setAclForTopic_(config);
        }
        if (common_1.util.isCustomType(config.destination, 'storage/bucket')) {
            await this.setAclForBucket_(config);
        }

        const reqOpts = {
            parent: 'projects/' + this.projectId,
            sink: extend({}, config, { name }),
        };
        delete reqOpts.sink.gaxOptions;
        await this.setProjectId(reqOpts);
        const [resp] = await this.configService.createSink(reqOpts, config.gaxOptions);
        const sink = this.sink(resp.name);
        sink.metadata = resp;
        return [sink, resp];
    }
```

#### Environment details

  - OS: Linux kali 5.2.0-kali3-amd64
  - Node.js version: v10.16.3
  - npm version: 6.13.6
  - `@google-cloud/logging` version: 7.0.0

#### Steps to reproduce

  1. Setup 2 GCP projects
  2. Create Service Account with "Log Configuration Writer" role
  3. Add above SA to secondary project (where SINK will be created)
  4. Create PubSub topic in first project
  5. Run above code in context of SA to create sink in secondary project
 
I hope this is quite enough to find the root cause and solve that issue in next release.

**Comment by bcoe** [comment]
@pdanysz thanks for the bug report. One thought as a work around, I think if you provided credentials explicitly using the variable `GOOGLE_APPLICATION_CREDENTIALS`, it should work.

Mind trying and letting me know if this unblocks you?

**Comment by pdanysz** [comment]
Hello @bcoe 
Unfortunately setting ``GOOGLE_APPLICATION_CREDENTIALS`` don't work - result is the same :)

**Comment by stephenplusplus** [comment]
@pdanysz I have created a PR to hopefully fix this issue: #686.
